util/pull/pull.go

```diff
@@ -123,7 +125,9 @@ func (p *Puller) Pull(ctx context.Context) (*Pulled, error) {
 		// Set any children labels for that content
 		childrenHandler = images.SetChildrenLabels(p.ContentStore, childrenHandler)
 		// Filter the childen by the platform
-		childrenHandler = images.FilterPlatforms(childrenHandler, platformStr)
+		childrenHandler = images.FilterPlatforms(childrenHandler, platform)
+		// Limit manifests pulled to the best match in an index
+		childrenHandler = images.LimitManifests(childrenHandler, platform, 1)
```

Comment:
@dmcgowan if `platform` is passed in here as well, isn't `FilterPlatforms` and `LimitManifests` redundant in terms of platforms filtering ?

Reply:
Right now `LimitManifests` only sorts and truncates, we could have it also do matching in the future since I believe the only object we care about doing the filtering on is the manifest index (`limitManifests` ONLY operates on manifest indexes).